lib/internal/console/constructor.js

```diff
@@ -366,7 +366,7 @@ Console.prototype.assert = function assert(expression, ...args) {
 Console.prototype.clear = function clear() {
   // It only makes sense to clear if _stdout is a TTY.
   // Otherwise, do nothing.
-  if (this._stdout.isTTY) {
+  if (process.env.TERM !== 'dumb' && this._stdout.isTTY) {
```

Comment:
Not sure about it. There are 2 cases when TERM="dumb" could apper:
- TERM is not set and parent program couldn't detect terminal features and set TERM="dumb"
- parent program intentionally set TERM="dumb" to limit output features

I guess the first case is rear and mostly it's the second one.
If we'll go with the second case as the relevant one, then warning isn't needed, because it's expected that the program(node) just will omit any escape codes in the output.

For the first case, there is already the detection code in `getColorDepth` and it silently defaults to COLORS_2 when can't detect the term.

Reply:
It's probably not necessary. I just thought it might be weird if someone actively calls the function and nothing happens. But I am fine without it.